q-mode.el

```diff
@@ -232,6 +232,7 @@
 (defun q-activate-buffer (buffer)
   "Set the `q-active-buffer' to the supplied BUFFER."
   (interactive "bactivate buffer: ")
+  (when (called-interactively-p 'any) (display-buffer buffer))
```

Comment:
replacing the current window seems unnatural, perhaps something like this works better:

`(when (called-interactively-p 'any) (display-buffer buffer '(nil (inhibit-same-window . t))))`

Reply:
> replacing the current window

`(when (called-interactively-p 'any) (display-buffer buffer))` doesn't replace in my environment ↓↓.

or how do you mean it does replace in yours? (is yours a plain-vanilla emacs and no other `display-buffer` related settings in `init.el` ?)

note: i tested in a spacemacs distribution emacs, which may or may not already have sane `display-buffer` preferences/defaults/magic set elsewhere. i found my `display-buffer-alist` is only set to irrelevant `(("^\\*Ilist\\*$" imenu-list-display-buffer))`, so unsure if or how spacemacs could interfere.

shouldn't we decouple the specific `display-buffer` preferences away from the `q-activate-buffer` function and delegate that to `display-buffer-alist` setting (à la [this](https://emacs.stackexchange.com/questions/12747/how-to-conditionally-reuse-the-current-window-to-display-a-buffer)) 

---
scenario 1:
(** - top buffer in current/active window; * - top buffer in other window)
```
+-----------------------------+----------------------------+
|"a.q"           Q-Script **  |"untitled"      Text    *   |
|"*q-localhost*" Q-Shell      |                            |
+-----------------------------+----------------------------+
```
after `(q-activate-buffer "*q-localhost*")`:
```
+-----------------------------+----------------------------+
|"a.q"           Q-Script **  |"*q-localhost*" Q-Shell *   |
|                             |"untitled"      Text        |
+-----------------------------+----------------------------+
```
---
scenario 2:
```
+-----------------------------+----------------------------+
|"a.q"           Q-Script **  |"untitled"      Text    *   |
|                             |"*q-localhost*" Q-Shell     |
+-----------------------------+----------------------------+
```
after `(q-activate-buffer "*q-localhost*")`:
```
+-----------------------------+----------------------------+
|"a.q"           Q-Script **  |"*q-localhost*" Q-Shell *   |
|                             |"untitled"      Text        |
+-----------------------------+----------------------------+
```
---
scenario 3:
```
+-----------------------------+
|"a.q"           Q-Script **  |
|"*q-localhost*" Q-Shell      |
+-----------------------------+
```
after `(q-activate-buffer "*q-localhost*")`:
```
+-----------------------------+----------------------------+
|"a.q"           Q-Script **  |"*q-localhost*" Q-Shell *   |
|                             |                            |
+-----------------------------+----------------------------+
```

Reply:
my `display-buffer-alist` has no customizations.  it is set to `nil`.

i've reproduced your examples, and they work as demonstrated.

this scenario, however, hides the existing buffer.

```
+-----------------------------+----------------------------+
|"a.q"           Q-Script **  |"b.q"           Q-Script *  |
|                             |                            |
+-----------------------------+----------------------------+
```

```
+-----------------------------+----------------------------+
|"*q-localhost*" Q-Shell **   |"b.q"           Q-Script *  |
|                             |                            |
+-----------------------------+----------------------------+
```
 nevertheless, i'll merge the PR.